src/test/unit/pages/LibraryPage.test.tsx

```diff
@@ -1,14 +1,37 @@
 import React from 'react';
-import { render, screen } from '@testing-library/react';
+import { act, render, screen } from '@testing-library/react';
 import LibraryPage from '../../../pages/LibraryPage';
+import * as libraryApi from "../../../api/libraryApi";
 
+const libraryStub = jest.spyOn(libraryApi, 'getLibraryWithGames');
 describe('Rendering LibraryPage', () => {
   beforeEach(() => {
-    render(<LibraryPage />);
+    libraryStub.mockResolvedValue({id: "123", 
+    name: "Want to Play", 
+    games: [{
+        id: "123", 
+        name: "Disco Elysium"}]
+      })
   });
 
-  it('renders library title', () => {
-    const title = screen.getByText(/My Library/i);
-    expect(title).toBeInTheDocument();
+  afterEach(() => {
+    jest.resetAllMocks();
+  })
+
+  describe('when loaded', () => {
+    beforeEach(async () => {
+      act(() => {
```

Comment:
I did something a little different in my PR, mine gives me a warning so this may be the approach we want to take. Regardless, we should keep it consistent

Reply:
Yep if you have state updates on your pages you'll need to wrap the render in an `act` block. I think it's best to minimize console/test warnings whenever we can as early as possible, to avoid the noise piling up